onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -550,6 +550,90 @@
         inputShape1.insert(inputShape1.begin(), outputShape.begin(), outputShape.end() - 2);
     }
 
+    void FusedMatMulShapeMapping(
+        std::vector<DimensionType>& inputShape0,
+        std::vector<DimensionType>& inputStride0,
+        std::vector<DimensionType>& inputShape1,
+        std::vector<DimensionType>& inputStride1,
+        std::vector<DimensionType>& outputShape)
+    {
+        // Get the padded input shapes, and undo the effect of padding removal from the output shape.
+        if (inputShape1.size() == 1)
+        {
+            inputShape1.push_back(1);
+            inputStride1.push_back(0);
+            outputShape.push_back(1);
+        }
+
+        if (inputShape0.size() == 1)
+        {
+            inputShape0.insert(inputShape0.begin(), 1);
+            inputStride0.insert(inputStride0.begin(), 0);
+            outputShape.insert(outputShape.end() - 1, 1);
+        }
+
+        auto broadcastedRank = std::max(inputShape0.size(), inputShape1.size());
+        inputShape0.insert(inputShape0.begin(), (broadcastedRank - inputShape0.size()), 1);
+        inputStride0.insert(inputStride0.begin(), (broadcastedRank - inputStride0.size()), 0);
+
+        inputShape1.insert(inputShape1.begin(), (broadcastedRank - inputShape1.size()), 1);
+        inputStride1.insert(inputStride1.begin(), (broadcastedRank - inputStride1.size()), 0);
+
+        // Broadcast the extra dimensions of each input, then add the truncated matrix dimensions.
+        BroadcastTensorShapeAndSetStrides(
+            gsl::make_span(inputShape0).subspan(0, broadcastedRank - 2),
+            gsl::make_span(inputStride0).subspan(0, broadcastedRank - 2),
+            gsl::make_span(inputShape1).subspan(0, broadcastedRank - 2),
+            gsl::make_span(inputStride1).subspan(0, broadcastedRank - 2)
+        );
+    }
+
+    std::pair<std::vector<uint32_t>, std::vector<uint32_t>> FusedMatMulSizeAndStride(
```

Comment:
>FusedMatMulSizeAndStride [](http://example.com/codeflow?start=60&length=24)

[naming] The function name is missing a fitting verb. Currently it sound like it fused something rather than getting something.

Propose `GetFusedMatMulSizesAndStrides`.